### PR TITLE
Adding 'get_simple_rules_ending_with' message and functionality for Tor

### PR DESCRIPTION
**Pull request by Hainish**
Closes https://github.com/EFForg/https-everywhere/issues/18983

### CODE REVIEW

File: chromium/background-scripts/rules.js [hunk 1]
```diff
@@ -722,6 +722,36 @@ RuleSets.prototype = {
...
+    let results;
+
+    if (this.wasm_rs) {
+      results = this.wasm_rs.get_simple_rules_ending_with(ending);
+    } else {
+      results = [];
+      for(let [host, rulesets] of this.targets) {
+        if (host.endsWith(ending) &&
+            rulesets.length == 1 &&
+            rulesets[0].active == true &&
```
Comment (cschanaj) [inline chromium/background-scripts/rules.js]:
```suggestion
            rulesets[0].active &&
```

File: chromium/background-scripts/rules.js [hunk 2]
```diff
@@ -722,6 +722,36 @@ RuleSets.prototype = {
...
+            rulesets[0].exclusions == null
+        ) {
+          for (let rule of rulesets[0].rules) {
+            if (rule.from_c.test("http://" + host + "/")) {
+              results.push({ host, from_regex: rule.from_c.toString(), to: rule.to, scope_regex: rulesets[0].scope.toString() });
```
Comment (cschanaj) [inline chromium/background-scripts/rules.js]: Both `from_regex` and `scope_regex` are String, IMHO it is quite confusing to have the `_regex` part in their names. They are the same as the `from` attribute from a ruleset and `scope` attribute from a ruleset channel, is there any reason not reusing the names?

Comment (Hainish) [reply chromium/background-scripts/rules.js]: Yeah, so `sendResponse` [must](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage) return a JSON-ifiable object, which is why we convert the RegExp to a string.

Given that this message is being used by an external project, I thought it safer to specify in the key that the `from` and `scope` should be used as a regular expression, thus the `_regex` suffix.

File: chromium/background-scripts/rules.js (hunk 1 above)
Comment (Hainish) [reply chromium/background-scripts/rules.js]: `active` should either be true or false in rulesets, and I'd like to maintain parity with the [`lib-core` implementation](https://github.com/EFForg/https-everywhere-lib-core/blob/master/src/rulesets.rs#L387) here, but this should probably be `===` instead.

**Pull request merged**